ensembl_genes/Red.py

```diff
@@ -197,42 +196,41 @@ def write_output(self):
         repeats_file = self.parse_repeats(self.param('rpt'),repeat_consensus_id,analysis_id)
 
         # insert repeat features
-        repeat_feature_query = "LOAD DATA LOCAL INFILE '"+repeats_file+"' \
-                                INTO TABLE repeat_feature \
-                                FIELDS TERMINATED BY '\\t' \
-                                LINES TERMINATED BY '\\n' \
-                                (seq_region_id,seq_region_start,seq_region_end,\
-                                 repeat_start,repeat_end,repeat_consensus_id,analysis_id)"
+        repeat_feature_query = f'LOAD DATA LOCAL INFILE "{repeats_file}"' \
+                               +"INTO TABLE repeat_feature \
+                                 FIELDS TERMINATED BY '\\t' \
+                                 LINES TERMINATED BY '\\n' \
+                                 (seq_region_id,seq_region_start,seq_region_end, \
+                                  repeat_start,repeat_end,repeat_consensus_id,analysis_id)"
         connection.execute(repeat_feature_query)
 
         # delete temporary directory and its contents
         try:
             shutil.rmtree(self.param('gnm'))
         except OSError as e:
-            print("Error: %s : %s" % (self.param('gnm'),e.strerror))
+            print(f'Error: {self.param("gnm")} : {e.strerror}')
 
     def parse_repeats(self,rpt,repeat_consensus_id,analysis_id):
         """ It parses the Red's program output and it converts it into
             a tsv file which can be loaded into an Ensembl core repeat_feature table. """
-        rpt_files = os.listdir(rpt)
+        rpt_files = [str(path) for path in Path(rpt).iterdir()]
         if not rpt_files:
             raise FileNotFoundError(errno.ENOENT,os.strerror(errno.ENOENT), \
-                                    "The repeats output directory "+rpt+" is empty.")
+                                    f'The repeats output directory {rpt} is empty.')
         elif len(rpt_files) > 1:
-            raise ValueError('The rpt output directory '+rpt+' contains more than 1 file.')
+            raise ValueError(f'The rpt output directory {rpt} contains more than 1 file.')
         elif not rpt_files[0].endswith('.rpt'):
-            raise ValueError('The file '+rpt_files[0]+' in the rpt output directory '+rpt+' \
-                              does not end with .rpt as required.')
+            raise ValueError(f'The file {rpt_files[0]} in {rpt} does not end with ".rpt".')
         else:
             # 1 file in rpt dir and ends with .rpt as expected
             # Red's rpt output file contains ">" which needs to be removed from each line
             # and we need to replace the seq region name with seq region id and
             # add some extra columns so it can be loaded directly
             seq_region = self.param('seq_region')
-            rpt_file = rpt+os.path.sep+rpt_files[0]
-            fixed_rpt_file = rpt_file+'.fixed'
+            rpt_file = Path(rpt_files[0])
```

Comment:
... so that you don't have to recreate a Path object here.

Reply:
I think I was trying to count the number of paths using Path but I could not find any way to do it simpler. I have had another look at this part of the code and I have removed the checks because they are not necessary any more since I delete the dir and create it again following the previous changes.